build.gradle.kts

```diff
@@ -14,13 +14,182 @@
+import java.time.Duration
+import java.time.temporal.ChronoUnit.SECONDS
+
 plugins {
-  id("com.hedera.hashgraph.aggregate-reports")
-  id("com.hedera.hashgraph.spotless-conventions")
-  id("com.hedera.hashgraph.spotless-kotlin-conventions")
+    id("com.hedera.hashgraph.aggregate-reports")
+    id("com.hedera.hashgraph.spotless-conventions")
+    id("com.hedera.hashgraph.spotless-kotlin-conventions")
+    id("com.bmuschko.docker-remote-api").version("9.1.0")
 }
 
 repositories {
-  mavenCentral()
-  maven { url = uri("https://oss.sonatype.org/content/repositories/snapshots") }
+    mavenCentral()
+    maven { url = uri("https://oss.sonatype.org/content/repositories/snapshots") }
+}
+
+var removeTempDockerFilesTask =
+        tasks.register<Delete>("removeTempDockerFiles") {
+            description =
+                    "Deletes all temp docker files that are copied in the root folder to create the docker image"
+            group = "docker"
+
+            delete(".env", ".dockerignore", "Dockerfile")
+        }
+
+tasks.clean { dependsOn(removeTempDockerFilesTask) }
+
+var updateDockerEnvTask =
+        tasks.register<Exec>("updateDockerEnv") {
+            description =
+                    "Creates the .env file in the docker folder that contains environment variables for docker"
+            group = "docker"
+
+            workingDir("./docker")
+            commandLine("./update-env.sh", project.version)
+        }
+
+tasks.register<Exec>("createDockerImage") {
+    description = "Creates the docker image of the services based on the current version"
+    group = "docker"
+
+    dependsOn(updateDockerEnvTask)
+    workingDir("./docker")
+    commandLine("./docker-build.sh", project.version)
+    finalizedBy(removeTempDockerFilesTask)
+}
+
+tasks.register<Exec>("startDockerContainers") {
+    description = "Starts docker containers of the services based on the current version"
+    group = "docker"
+
+    dependsOn(updateDockerEnvTask)
+    workingDir("./docker")
+    commandLine("docker-compose", "up")
+}
+
+tasks.register<Exec>("stopDockerContainers") {
+    description = "Stops running docker containers of the services"
+    group = "docker"
+
+    dependsOn(updateDockerEnvTask)
+    workingDir("./docker")
+    commandLine("docker-compose", "stop")
 }
+
+var buildPrometheusImageTask =
+        tasks.register<DockerBuildImage>("buildPrometheusImage") {
+            description = "Builds Prometheus Docker image"
+            group = "docker intern"
+
+            inputDir.set(file("docker/prometheus"))
+            images.add("swirlds/platform-prometheus:latest")
+        }
+
+var buildGrafanaImageTask =
+        tasks.register<DockerBuildImage>("buildGrafanaImage") {
+            description = "Builds Grafana Docker image"
+            group = "docker intern"
+
+            inputDir.set(file("docker/grafana"))
+            images.add("swirlds/services-grafana:latest")
+        }
```

Comment:
Should not be needed at all, since a bind mount will suffice.  This adds increased start up times and makes it hard to quickly test changes to the dashboards.